### PR TITLE
Balder/refactor docentry

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR
Probably best to look at the end result. I just made the changes in steps to have control at each point.

### CODE REVIEW

File: storage/src/vespa/storage/visiting/reindexing_visitor.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ class ReindexingVisitor : public Visitor {
     ~ReindexingVisitor() override = default;
 
 private:
-    void handleDocuments(const document::BucketId&, std::vector<spi::DocEntry::UP>&, HitCounter&) override;
+    void handleDocuments(const document::BucketId&, DocEntryList&, HitCounter&) override;
```
Comment (vekterli) [inline storage/src/vespa/storage/visiting/reindexing_visitor.h:23]: Consider updating the parameter type signature in the .cpp file as well

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/visiting/reindexing_visitor.h (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/visiting/reindexing_visitor.h:23]: Done

File: persistence/src/vespa/persistence/spi/docentry.h [hunk 2]
```diff
@@ -14,80 +14,59 @@
...
-    const Document* getDocument() const { return _document.get(); }
-    const DocumentId* getDocumentId() const;
-    DocumentUP releaseDocument();
-    bool isRemove() const { return (_metaFlags & REMOVE_ENTRY); }
+    DocEntry(const DocEntry &) = delete;
+    DocEntry & operator=(const DocEntry &) = delete;
+    DocEntry(DocEntry &&) = delete;
+    DocEntry & operator=(DocEntry &&) = delete;
+    virtual ~DocEntry();
+    bool isRemove() const { return (_metaFlags == DocumentMetaFlags::REMOVE_ENTRY); }
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply persistence/src/vespa/persistence/spi/docentry.h:37]: Agree, DocumentMetaEnum, getMetaEnum

**Pull request merged**